src/promptflow/promptflow/_cli/_pf/_tool.py

```diff
@@ -65,9 +68,38 @@ def add_parser_init_tool(subparsers):
     )
 
 
+def add_parser_list_tool(subparsers):
+    """Add tool list parser to the pf tool subparsers."""
+    epilog = """
+Examples:
+
+# List all package tool in the environment:
+pf tool list
+# List all package tool and code tool in the flow:
+pf tool list --flow flow-path
+"""  # noqa: E501
+    add_param_flow = lambda parser: parser.add_argument(  # noqa: E731
+        "--flow", type=str, help="the flow directory"
```

Comment:
Do we also support specifying flow dag file?

Reply:
Yes, it also support specifying flow dag file. But now all flow description in CLI points to flow folder.